### PR TITLE
Add app.bbsync.BlackboardSync

### PR DESCRIPTION
**Pull request by jacobszpz**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
No patches

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: app.bbsync.BlackboardSync.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    </screenshot>
+    <screenshot>
+      <image type="source">https://bbsync.app/assets/screenshots/BBS_Login.png</image>
+    </screenshot>
+    <screenshot>
+      <image type="source">https://bbsync.app/assets/screenshots/BBS_Notification.png</image>
+    </screenshot>
+  <screenshots/>
+
+  <url type="homepage">https://github.com/jacobszpz/BlackboardSync</url>
```
Comment (hfiguiere) [inline app.bbsync.BlackboardSync.metainfo.xml:32]: why isn't the `bbsync.app`?

File: app.bbsync.BlackboardSync.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+</component>
```
Comment (hfiguiere) [inline app.bbsync.BlackboardSync.metainfo.xml:33]: missing required `releases`, `content_rating`

File: app.bbsync.BlackboardSync.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+<!-- Copyright 2023 Jacob Sanchez -->
+<component type="desktop-application">
+  <id>app.bbsync.BlackboardSync</id>
+  <metadata_license>FSFAP</metadata_license>
+  <project_license>GPL-2.0</project_license>
+  <content_rating type="oars-1.1" />
+  <launchable type="desktop-id">app.bbsync.BlackboardSync.desktop</launchable>
+  <name>BlackboardSync</name>
+  <summary>Download your Blackboard Learn content to your device.</summary>
+  <summary xml:lang="es">Descarga contenido de Blackboard Learn a tu dispositivo.</summary>
```
Comment (hfiguiere) [inline app.bbsync.BlackboardSync.metainfo.xml:11]:
```suggestion
  <summary>Download your Blackboard Learn content to your device</summary>
  <summary xml:lang="es">Descarga contenido de Blackboard Learn a tu dispositivo</summary>
```

File: app.bbsync.BlackboardSync.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        buildsystem: simple
+        build-commands:
+          - install -Dm644 app.bbsync.BlackboardSync.metainfo.xml -t "${FLATPAK_DEST}/share/metainfo"
+          - install -Dm644 app.bbsync.BlackboardSync.desktop -t "${FLATPAK_DEST}/share/applications"
+          - install -Dm644 /app/lib/python3*/site-packages/blackboard_sync/assets/logo.png "${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/app.bbsync.BlackboardSync.png"
+        sources:
+          - type: file
+            path: app.bbsync.BlackboardSync.metainfo.xml
+          - type: file
+            path: app.bbsync.BlackboardSync.desktop
```
Comment (hfiguiere) [inline app.bbsync.BlackboardSync.yml:48]:
```suggestion
  - name: BlackboardSync
    buildsystem: simple
    modules:
      - python-maturin/python-maturin.json
      - pydantic-core/pydantic-core.json
      - python3-requirements.json
      - name: python-blackboardsync
        buildsystem: simple
        build-commands:
          - pip3 install --verbose --exists-action=i --no-index --prefix=${FLATPAK_DEST} --no-deps . --no-build-isolation
        sources:
          - blackboardsync-source.json
    build-commands:
      - install -Dm644 app.bbsync.BlackboardSync.metainfo.xml -t "${FLATPAK_DEST}/share/metainfo"
      - install -Dm644 app.bbsync.BlackboardSync.desktop -t "${FLATPAK_DEST}/share/applications"
      - install -Dm644 /app/lib/python3*/site-packages/blackboard_sync/assets/logo.png "${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/app.bbsync.BlackboardSync.png"
    sources:
      - type: file
        path: app.bbsync.BlackboardSync.metainfo.xml
      - type: file
        path: app.bbsync.BlackboardSync.desktop
```

File: pydantic-core/pydantic-core-source.json
Comment (hfiguiere) [inline pydantic-core/pydantic-core-source.json:1]: this has no reason to be a separate file. it's hard to maintain

File: pydantic-core/pydantic-core.json
Comment (hfiguiere) [inline pydantic-core/pydantic-core.json:1]: this has no reason to be a separate file. it's hard to maintain

File: python-maturin/python-maturin-source.json
Comment (hfiguiere) [inline python-maturin/python-maturin-source.json:1]: this has no reason to be a separate file. it's hard to maintain

File: python-maturin/python-maturin.json
Comment (hfiguiere) [inline python-maturin/python-maturin.json:1]: this has no reason to be a separate file. it's hard to maintain

File: app.bbsync.BlackboardSync.desktop
Comment (hfiguiere) [inline app.bbsync.BlackboardSync.desktop:1]: this file should be managed upstream

File: app.bbsync.BlackboardSync.metainfo.xml
Comment (hfiguiere) [inline app.bbsync.BlackboardSync.metainfo.xml:1]: this file should be managed upstream

File: app.bbsync.BlackboardSync.metainfo.xml (hunk 1 above)
Comment (jacobszpz) [reply app.bbsync.BlackboardSync.metainfo.xml:32]: I can make this change

File: app.bbsync.BlackboardSync.metainfo.xml (hunk 2 above)
Comment (jacobszpz) [reply app.bbsync.BlackboardSync.metainfo.xml:33]: content rating is there, at the top. I will add releases

File: app.bbsync.BlackboardSync.desktop
Comment (jacobszpz) [reply app.bbsync.BlackboardSync.desktop:1]: can put it upstream

File: app.bbsync.BlackboardSync.metainfo.xml
Comment (jacobszpz) [reply app.bbsync.BlackboardSync.metainfo.xml:1]: also can put it upstream

**Comment by jacobszpz** [comment]
I have fixed most things, save that I prefer having the subdirs python-maturin and pydantic-core to keep those submodules separate, as these dependencies have a completely different build process than the rest. Is this something you can allow?

**Comment by hfiguiere** [comment]
The thing is when maintainer go AWOL, people like me end up picking up the pieces. so making it unpleasant by splitting the manifest like that (only GENERATED and shared-modules manifest are meant to be like that) make it unpleasant to us.

Regrettably some people have made that choice.

**Comment by hfiguiere** [comment]
bot, build app.bbsync.BlackboardSync

**Comment by hfiguiere** [comment]
build fail because of a pip error:
```
  /usr/lib/python3.10/site-packages/setuptools/command/egg_info.py:131: SetuptoolsDeprecationWarning: Invalid version: '0.9.12-alpha.1.2'.
```

**Comment by jacobszpz** [comment]
I have looked into what you say and I have found a way to make it work. I will update in a few days with a combined manifest (save for generated sources) and as for the pip error that's only because of the version format not being supported by pip, a new build using a different format should work. I will update when done.

File: app.bbsync.BlackboardSync.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    buildsystem: simple
+    modules:
+      - python-maturin/python-maturin.json
+      - pydantic-core/pydantic-core.json
+      - python3-requirements.json
+      - name: python-blackboardsync
+        buildsystem: simple
+        build-commands:
+          - pip3 install --verbose --exists-action=i --no-index --prefix=${FLATPAK_DEST} --no-deps . --no-build-isolation
+          - ls .
```
Comment (bbhtt) [inline app.bbsync.BlackboardSync.yml:34]:
```suggestion

```

Leftover

**Comment by bbhtt** [comment]
Fix this upstream:

```
app.bbsync.BlackboardSync.metainfo.xml: failed to parse app.bbsync.BlackboardSync.metainfo.xml: Error on line 38 char 13: Element “component” was closed, but the currently open element is “releases”
```

**Comment by jacobszpz** [comment]
We should be good to go, save for the icon. Which size / sizes would be best?

**Comment by hfiguiere** [comment]
128x128 is the minimum size. 256x256 or 512x512 can work. Best is just a SVG.

**Comment by jacobszpz** [comment]
Done

**Comment by bbhtt** [comment]
bot, build app.bbsync.BlackboardSync

**Comment by jacobszpz** [comment]
It works as expected.

**Pull request closed without merging**